src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -753,21 +765,49 @@ private void OnClusterTransactionCompletion(ClusterTransactionCommand.SingleClus
             {
                 var index = command.Index;
                 var options = mergedCommands.Options[index];
-                Task indexTask = null;
+                var taskResult = new ClusterTransactionResult
+                {
+                    GeneratedResult = mergedCommands.Replies[index],
+                };
                 if (options.WaitForIndexesTimeout != null)
                 {
-                    indexTask = BatchHandlerProcessorForBulkDocs.WaitForIndexesAsync(this, options.WaitForIndexesTimeout.Value,
+                    var indexTask = BatchHandlerProcessorForBulkDocs.WaitForIndexesAsync(this, options.WaitForIndexesTimeout.Value,
                         options.SpecifiedIndexesQueryString, options.WaitForIndexThrow,
                             mergedCommands.LastDocumentEtag, mergedCommands.LastTombstoneEtag, mergedCommands.ModifiedCollections);
-                }
 
-                var result = new ClusterTransactionCompletionResult
+                    var removeTask = ServerStore.Cluster.ClusterTransactionWaiter.CreateTask(command.Options.TaskId);
+
+                    /*
+                        the remove task that we are creating here is relevant only for a failover with wait for index:
+                        We need the task only when we do a failover during cluster transaction (batch handler) with 'wait for index'.
+                        Only then we will want the task that created here, because we'll want to 'wait for index' in the new node after the failover.
+                    */
+
+                    indexTask.ContinueWith(t =>
```

Comment:
Avoid throwing here, you can check via the `IsCompletedSucessfully`

Reply:
Done, I preferred to check if it's faulted